govcd/lbserverpool.go

```diff
@@ -0,0 +1,177 @@
+package govcd
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// CreateLBServerPool
+func (eGW *EdgeGateway) CreateLBServerPool(lbPoolConfig *types.LBPool) (*types.LBPool, error) {
+	if err := validateCreateLBServerPool(lbPoolConfig); err != nil {
+		return nil, err
+	}
+
+	httpPath, err := eGW.buildProxiedEdgeEndpointURL(types.LBServerPoolPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
+	}
+	// We expect to get http.StatusCreated or if not an error of type types.NSXError
+	resp, err := eGW.client.ExecuteRequestWithCustomError(httpPath, http.MethodPost, types.AnyXMLMime,
+		"error creating load balancer server pool: %s", lbPoolConfig, &types.NSXError{})
+	if err != nil {
+		return nil, err
+	}
+	location := resp.Header.Get("Location")
+
+	// Last element in location header is the server pool ID
+	// i.e. Location: [/network/edges/edge-3/loadbalancer/config/pools/pool-7]
+	if location == "" {
+		return nil, fmt.Errorf("unable to retrieve ID for new load balancer server pool with name %s", lbPoolConfig.Name)
+	}
+	splitLocation := strings.Split(location, "/")
+	lbPoolID := splitLocation[len(splitLocation)-1]
+	readPool, err := eGW.ReadLBServerPool(&types.LBPool{ID: lbPoolID})
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve lb server pool with ID (%s) after creation: %s", readPool.ID, err)
+	}
+	return readPool, nil
+}
+
+func (eGW *EdgeGateway) ReadLBServerPool(lbPoolConfig *types.LBPool) (*types.LBPool, error) {
```

Comment:
A function comment would be nice, esp. for other users than Terraform.

Reply:
Done.